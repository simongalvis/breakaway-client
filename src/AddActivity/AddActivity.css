.AddActivity{
    display: flex;
    flex-direction: column;
}