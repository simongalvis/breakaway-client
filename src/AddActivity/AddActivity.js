import React from 'react';
import ApiContext from '../ApiContext';
import DashboardNavBar from '../DashboardNavBar/DashboardNavBar';
import './AddActivity.css'


class AddActivity extends React.Component{
static contextType = ApiContext;

    handleSubmit = e =>{
        e.preventDefault();
        const { titleInput, durationInput, distanceInput, descriptionInput } = e.target;

    
       
        const activity = {
            name: this.context.loggedUser['fullname'],
            user_id: this.context.loggedUser['id'],
            title: titleInput.value,
            duration: durationInput.value,
            distance: distanceInput.value,
            description: descriptionInput.value
        }
        
        

        this.context.addActivity(activity)

        setTimeout( () => this.props.history.push('/activities'), 500)
        
        
    }
    

    

    render(){
        return(
            <div className='AddActivity'>
                <DashboardNavBar/>
                
                <div className='addActivityFormContainer'>
                <h2 id='loginheading'>Add Activity</h2>
                <form onSubmit={this.handleSubmit}>
                    <label for='titleInput'><b>Title:</b></label>
                    <input  type='text' id='titleInput' name='titleInput' placeholder='Title' />
                    <label for='durationInput'><b>Duration:</b> </label>
                    <input  type='number' id='durationInput' name='durationInput' placeholder='Duration(minutes)' required/>
                    <label for='distanceInput'><b>Distance:</b> </label>
                    <input  type='number' id='distanceInput' name='distanceInput' placeholder='Distance(miles)' required/>
                    <label for='descriptionInput'><b>Notes:</b> </label>
                    <input  type='text' id='descriptionInput' name='descriptionInput' placeholder='Notes(Ex: Felt great after the first mile)' required/>
                    <button type='submit' id='activity-submit-button'>Submit</button>
                   
                    
                </form>
                </div>
                    
                    
                    
                
                
            </div>
        
        )
    }
}

export default AddActivity;