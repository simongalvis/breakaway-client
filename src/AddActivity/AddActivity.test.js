import React from 'react'
import ReactDOM from 'react-dom'
import AddActivity from './AddActivity'
import renderer from 'react-test-renderer';

describe(`AddActivity component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<AddActivity/>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<AddActivity/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
