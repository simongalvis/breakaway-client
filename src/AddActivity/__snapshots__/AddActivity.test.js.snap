// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivity component Renders UI as expected 1`] = `
<div
  className="AddActivity"
>
  <header>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Blue breakaway logo"
        id="main-logo"
        src="breakaway-logo.png"
      />
    </a>
    <a
      aria-current={null}
      href="/new-activity"
      onClick={[Function]}
    >
      <button
        className="nav-button"
        id="newactivity-breakaway-button"
        value="new activity"
      >
        <b>
           +Breakaway!
        </b>
          
      </button>
    </a>
  </header>
  <div
    className="addActivityFormContainer"
  >
    <h2
      id="loginheading"
    >
      Add Activity
    </h2>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="titleInput"
      >
        <b>
          Title:
        </b>
      </label>
      <input
        id="titleInput"
        name="titleInput"
        placeholder="Title"
        type="text"
      />
      <label
        htmlFor="durationInput"
      >
        <b>
          Duration:
        </b>
         
      </label>
      <input
        id="durationInput"
        name="durationInput"
        placeholder="Duration(minutes)"
        required={true}
        type="number"
      />
      <label
        htmlFor="distanceInput"
      >
        <b>
          Distance:
        </b>
         
      </label>
      <input
        id="distanceInput"
        name="distanceInput"
        placeholder="Distance(miles)"
        required={true}
        type="number"
      />
      <label
        htmlFor="descriptionInput"
      >
        <b>
          Notes:
        </b>
         
      </label>
      <input
        id="descriptionInput"
        name="descriptionInput"
        placeholder="Notes(Ex: Felt great after the first mile)"
        required={true}
        type="text"
      />
      <button
        id="activity-submit-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
